File: backend/src/main/java/com/bankingsystem/service/AdminService.java
package com.bankingsystem.service;

import com.bankingsystem.dto.TransactionDTO;
import com.bankingsystem.dto.UserDTO;
import java.util.List;
import java.util.Map;

public interface AdminService {
    List<UserDTO> getAllUsers();
    UserDTO getUserById(Long userId);
    void approveUser(Long userId);
    void freezeUser(Long userId);
    void activateUser(Long userId);
    void deactivateUser(Long userId);
    void deleteUser(Long userId);
    Map<String, Object> getDashboardStatistics();
    List<TransactionDTO> getAllTransactions();
    Map<String, Object> getReportData();
}

File: frontend/src/services/AdminService.js
import api from './api'

const AdminService = {
  // User management
  getUsers: () => api.get('/api/admin/users'),
  getUserById: (id) => api.get(`/api/admin/users/${id}`),
  approveUser: (id) => api.put(`/api/admin/users/${id}/approve`),
  freezeUser: (id) => api.put(`/api/admin/users/${id}/freeze`),
  activateUser: (id) => api.put(`/api/admin/users/${id}/activate`),
  deactivateUser: (id) => api.put(`/api/admin/users/${id}/deactivate`),
  deleteUser: (id) => api.delete(`/api/admin/users/${id}`),

  // Dashboard
  getStats: () => api.get('/api/admin/dashboard/statistics'),

  // Loans
  getAllLoans: () => api.get('/api/admin/loans'),
  updateLoanStatus: (id, status) => api.patch(`/api/loans/${id}/status`, { status }),

  // Transactions
  getAllTransactions: () => api.get('/api/admin/transactions'),

  // Reports
  getReports: () => api.get('/api/admin/reports'),

  // Accounts
  getAllAccounts: () => api.get('/api/admin/accounts'),

  // Contact
  getContactMessages: () => api.get('/api/contact'),
  markMessageRead: (id) => api.put(`/api/contact/${id}/read`),
}

export default AdminService
